src/DropKit.ts

```diff
@@ -138,29 +160,74 @@ export default class DropKit {
   }
 
   async mint(quantity: number): Promise<void> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
+    try {
+      // safety check
+      quantity = Number(Math.min(quantity, await this.maxPerMint()))
+
+      const presaleActive = await this.presaleActive()
+      const saleActive = await this.saleActive()
+
+      const tokensCount = await this.walletTokensCount()
+      const maxPerWallet = await this.maxPerWallet()
+
+      if (tokensCount >= maxPerWallet) {
+        throw new Error(
+          `You can't mint more than ${maxPerWallet} tokens on your wallet`
+        )
+      }
+
+      if (!saleActive && !presaleActive) {
+        throw new Error('Collection is not active')
+      }
+
+      const price = await this.price()
+      const amount = ethers.utils.parseEther(price.toString()).mul(quantity)
+
+      // Presale minting
+      if (presaleActive) {
+        // Backwards compatibility with v2 contracts:
+        // If the public sale is not active, we can still try mint with the presale
+        await this._presaleMint(quantity, amount)
+        return
+      }
+
+      // Regular minting
+      await this._mint(quantity, amount)
+    } catch (error) {
+      handleError(error as EthereumRpcError<unknown>)
     }
-    const price = await this.price()
+  }
 
-    // Presale mint
-    try {
-      const data = await this.generateProof()
-      const results = await this.contract.presaleMint(quantity, data.proof, {
-        value: ethers.utils.parseEther(price.toString()).mul(quantity),
-      })
-      await results.wait()
-
-      return
-    } catch (err) {
-      console.log(err)
+  private async _mint(
+    quantity: number,
+    amount: ethers.BigNumber
+  ): Promise<void> {
+    const trx = await this.contract?.mint(quantity, {
+      value: amount,
+    })
+
+    await trx.wait()
```

Comment:
we should `return trx.wait()` instead. that way it will return a promise

Reply:
Added better typings support, now `_mint` and `_presaleMint` returns `Promise<ContractReceipt>`
 see: https://github.com/niftykit-inc/dropkit.js/pull/6/commits/7215b584295fb3428cd1e99daf45fbc0e3afddfb